src/app_service/service/message/dto/link_preview.nim

```diff
@@ -75,10 +75,10 @@ proc `$`*(self: LinkPreview): string =
 proc `%`*(self: LinkPreview): JsonNode =
   result = %* {
     "url": self.url,
-    "standardPreview": %self.standardPreview,
-    "contactPreview": %self.statusContactPreview,
-    "communityPreview": %self.statusCommunityPreview,
-    "channelPreview": %self.statusCommunityChannelPreview
+    "standardPreview": if self.standardPreview != nil: %self.standardPreview else: newJNull(),
```

Comment:
Maybe its worth to introduce some helper function, like this?
```
proc toJsonOrNull[T](value: ref T): JsonNode =
  if value != nil:
    return %value
  else:
    return newJNull()
```

Reply:
Good idea, will try as follow-up